File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.rs
impl Solution {
    pub fn search_matrix(matrix: Vec<Vec<i32>>, target: i32) -> bool {
        let (m, n) = (matrix.len(), matrix[0].len());

        let mut row = m - 1;
        let mut col = 0;
        
            while col < n && row >=0  {
                
                if matrix[row][col] < target{
                    col+= 1;
                }else if matrix[row][col] > target{
                    if row == 0{
                        return false;
                    }
                    row-=1;
                }else{
                    return true;
                }
            }
        
        false
        
    }
}

File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        (m, n) = (len(matrix), len(matrix[0]))
        (i, j) = (0, n - 1)
        
        while i < m and j >= 0:
            print(i, j)
            if matrix[i][j] < target:
                i+=1
            elif matrix[i][j] > target:
                j-=1
            else:
                return True
        
        return False